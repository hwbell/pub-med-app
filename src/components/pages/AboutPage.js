import React from 'react';
import '../../App.scss';
import 'bootstrap/dist/css/bootstrap.css';
import { aboutPageText } from "../../assets/text";
// components
import Header from '../Header';
import OutlinedText from '../OutlinedText';

// elements

// styling
// import ReactCSSTransitionGroup from 'react-addons-css-transition-group';

class AboutPage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      // 
    };
  }

  componentDidMount() {
    // 
  }

  render() {
    return (
      <div className="about-page page">
        <div className="glass page-content">

          <Header
            class="heading"
            title={"About PubMed Central"}
            subtitle={"medical & life sciences literature for the public"}
          />

          <div className="outline" style={styles.content}>

            <p className="paragraph" style={styles.text}>{aboutPageText.full.top}</p>

          </div>

          <img style={{ margin: '15px', borderRadius: '10px' }} src={require('../../assets/about-page.png')}></img>

          <OutlinedText
            text={aboutPageText.full.bottom}
          />

        </div>
      </div>
    );
  }
}

const styles = {
  main: {
    width: '100%'
  }
}

export default AboutPage;
