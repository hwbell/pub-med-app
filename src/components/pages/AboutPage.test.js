import React from 'react';
import AboutPage from './AboutPage';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

// tests
it('renders without crashing', async () => {
  let wrapper = shallow(<AboutPage />);
  // wrapper.update();

});

it('renders correctly', async () => {
  const tree = await renderer
    .create(<AboutPage />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains the corect elements', () => {
  let wrapper = mount(<AboutPage />);
  expect(wrapper.find('.page').length).toEqual(1);
  expect(wrapper.find('.glass').length).toEqual(1);

});
