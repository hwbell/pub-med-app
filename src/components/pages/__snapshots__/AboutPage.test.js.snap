// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="about-page page"
>
  <div
    className="glass page-content"
  >
    <div
      className="heading"
    >
      <h2
        className="title"
      >
        About PubMed Central
      </h2>
      <h4
        className="subtitle"
      >
        medical & life sciences literature for the public
      </h4>
    </div>
    <div
      className="outline"
    >
      <p
        className="paragraph"
      >
        From Wikipedia, the free encyclopedia:

      PubMed Central (PMC) is a free digital repository that archives publicly 
      accessible full-text scholarly articles that have been published within 
      the biomedical and life sciences journal literature. As one of the major 
      research databases within the suite of resources that have been developed 
      by the National Center for Biotechnology Information (NCBI), PubMed Central 
      is much more than just a document repository. Submissions into PMC undergo
      an indexing and formatting procedure which results in enhanced metadata, 
      medical ontology, and unique identifiers which all enrich the XML structured 
      data for each article on deposit.[1] Content within PMC can easily be 
      interlinked to many other NCBI databases and accessed via Entrez search 
      and retrieval systems, further enhancing the public's ability to freely 
      discover, read and build upon this portfolio of biomedical knowledge.[2]
      

      

      
      PubMed Central is very distinct from PubMed.[3] PubMed Central is a free 
      digital archive of full articles, accessible to anyone from anywhere via 
      a web browser (with varying provisions for reuse). Conversely, although 
      PubMed is a searchable database of biomedical citations and abstracts, 
      the full-text article physically resides elsewhere (in print or online, 
      free or behind a subscriber paywall).
      </p>
    </div>
    <img
      src="about-page.png"
      style={
        Object {
          "borderRadius": "10px",
          "margin": "15px",
        }
      }
    />
    <div
      className="outline"
      style={
        Object {
          "margin": "10px auto",
          "width": "100%",
        }
      }
    >
      <p
        className="paragraph"
        style={
          Object {
            "alignSelf": "",
            "paddingTop": "10px",
          }
        }
      >
        As of December 2018, the PMC archive contained over 5.2 million 
    articles,[4] with contributions coming directly from publishers or authors 
    depositing their own manuscripts into the repository per the NIH Public 
    Access Policy. Older data shows that from Jan 2013 to Jan 2014 
    author-initiated deposits exceeded 103,000 papers during this 12-month 
    period.[5] PMC also identifies about 4,000 journals which now participate 
    in some capacity to automatically deposit their published content into the 
    PMC repository.[6] Some participating publishers will delay the release of 
    their articles on PubMed Central for a set time after publication, this is 
    often referred to as an "embargo period", and can range from a few months 
    to a few years depending on the journal. (Embargoes of six to twelve months 
      are the most common.) However, PubMed Central is a key example of 
      "systematic external distribution by a third party"[7] which is still 
      prohibited by the contributor agreements of many publishers.

    Launched in February 2000, the repository has grown rapidly as the NIH 
    Public Access Policy is designed to make all research funded by the National 
    Institutes of Health (NIH) freely accessible to anyone, and, in addition, 
    many publishers are working cooperatively with the NIH to provide free access 
    to their works. In late 2007, the Consolidated Appropriations Act of 
    2008 (H.R. 2764) was signed into law and included a provision requiring 
    the NIH to modify its policies and require inclusion into PubMed Central 
    complete electronic copies of their peer-reviewed research and findings 
    from NIH-funded research. These articles are required to be included within 
    12 months of publication. This is the first time the US government has 
    required an agency to provide open access to research and is an evolution 
    from the 2005 policy, in which the NIH asked researchers to voluntarily 
    add their research to PubMed Central.[8]
    
    A UK version of the PubMed Central system, UK PubMed Central (UKPMC), has been 
    developed by the Wellcome Trust and the British Library as part of a 
    nine-strong group of UK research funders. This system went live in January 
    2007. On 1 November 2012, it became Europe PubMed Central. The Canadian 
    member of the PubMed Central International network, PubMed Central Canada, 
    was launched in October 2009.
    
    The National Library of Medicine "NLM Journal Publishing Tag Set" journal 
    article markup language is freely available.[9] The Association of Learned 
    and Professional Society Publishers comments that "it is likely to become 
    the standard for preparing scholarly content for both books and journals".[10] 
    A related DTD is available for books.[11] The Library of Congress and the 
    British Library have announced support for the NLM DTD.[12] It has also 
    been popular with journal service providers.[13]
    
    With the release of public access plans for many agencies beyond NIH, PMC 
    is in the process of becoming the repository for a wider variety of 
    articles.[14] This includes NASA content, with the interface branded as 
    "PubSpace".[15][16]
    
    Articles are sent to PubMed Central by publishers in XML or SGML, using 
    a variety of article DTDs. Older and larger publishers may have their own 
    established in-house DTDs, but many publishers use the NLM Journal 
    Publishing DTD (see above).
    
    Received articles are converted via XSLT to the very similar NLM Archiving 
    and Interchange DTD. This process may reveal errors that are reported back 
    to the publisher for correction. Graphics are also converted to standard 
    formats and sizes. The original and converted forms are archived. The 
    converted form is moved into a relational database, along with associated 
    files for graphics, multimedia, or other associated data. Many publishers 
    also provide PDF of their articles, and these are made available without 
    change.[17]
    
    Bibliographic citations are parsed and automatically linked to the relevant 
    abstracts in PubMed, articles in PubMed Central, and resources on publishers' 
    Web sites. PubMed links also lead to PubMed Central. Unresolvable references, 
    such as to journals or particular articles not yet available at one of these 
    sources, are tracked in the database and automatically come "live" when the 
    resources become available.
    
    An in-house indexing system provides search capability, and is aware of 
    biological and medical terminology, such as generic vs. proprietary drug 
    names, and alternate names for organisms, diseases and anatomical parts.
    
    When a user accesses a journal issue, a table of contents is automatically 
    generated by retrieving all articles, letters, editorials, etc. for that 
    issue. When an actual item such as an article is reached, PubMed Central 
    converts the NLM markup to HTML for delivery, and provides links to related 
    data objects. This is feasible because the variety of incoming data has first 
    been converted to standard DTDs and graphic formats.
    
    In a separate submission stream, NIH-funded authors may deposit articles 
    into PubMed Central using the NIH Manuscript Submission (NIHMS). Articles 
    thus submitted typically go through XML markup in order to be converted to 
    NLM DTD.
      </p>
    </div>
  </div>
</div>
`;
